controllers/provider-azure/pkg/controller/worker/machines.go

```diff
@@ -135,52 +146,93 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			image["urn"] = *urn
 		}
 
-		machineClassSpec := map[string]interface{}{
-			"region":            w.worker.Spec.Region,
-			"resourceGroup":     infrastructureStatus.ResourceGroup.Name,
-			"vnetName":          infrastructureStatus.Networks.VNet.Name,
-			"subnetName":        nodesSubnet.Name,
-			"availabilitySetID": nodesAvailabilitySet.ID,
-			"tags": map[string]interface{}{
-				"Name": w.worker.Namespace,
-				fmt.Sprintf("kubernetes.io-cluster-%s", w.worker.Namespace): "1",
-				"kubernetes.io-role-node":                                   "1",
-			},
-			"secret": map[string]interface{}{
-				"cloudConfig": string(pool.UserData),
-			},
-			"machineType":  pool.MachineType,
-			"image":        image,
-			"volumeSize":   volumeSize,
-			"sshPublicKey": string(w.worker.Spec.SSHPublicKey),
+		generateMachineClassAndDeployment := func(zone *zoneInfo, availabilitySetID *string) (worker.MachineDeployment, map[string]interface{}) {
```

Comment:
I tried to apply DRY. Just `if-else` branches would produce a lot of redundancy.
I choose the closure instead of regular function because we need only to pass the zone or availabilitySet information as parameters for the function...
I'm also fine with converting this to a regular function but then we need to pass as well the `pool`, `infrastructureStatus`, `image`, `volumeSize`, `shootVersionMajorMinor` ...

Reply:
OK